apps/google-analytics-4/frontend/src/services/api.ts

```diff
@@ -133,9 +139,12 @@ export class Api {
     return await fetchFromApi<Credentials>(
       this.requestUrl('api/credentials'),
       ZCredentials,
-      this.appDefinitionId,
+      this.contentfulContext.app!,
```

Comment:
@jsdalton / @ryunsong-contentful maybe worth pairing on this `!` typescript piece to see if one of you two have a suggestion for me 😄

Reply:
I guess I wonder why `app` is an optional field in https://github.com/contentful/apps/blob/master/apps/google-analytics-4/frontend/src/types.ts#L8 ? That's where the smell is coming from in my opinion. I _think_ all of those should be "required".

Reply:
yeah so I will revisit this but I was basing this on:

```
export interface ContentfulContext extends IdsAPI {
  app?: string;
  contentType?: string;
  entry?: string;
  environment: string;
  environmentAlias?: string;
  field?: string;
  location?: string;
  organization: string;
  space: string;
  user: string;
}
```

and had actually attempted to make this `extend` IdsAPI originally.

Reply:
The problem is that when you attempt to use the `sdk.ids` and have this misaligned – sdk.ids is a type `Omit<IdsAPI, EntryScopedIds>` – happy to pair w/u a bit on this detail as I was going in circles w/this – there's probably a "safe" coercion we can apply at the right spot here

Reply:
Also:

> I guess I wonder why app is an optional field in https://github.com/contentful/apps/blob/master/apps/google-analytics-4/frontend/src/types.ts#L8 ? That's where the smell is coming from in my opinion. I think all of those should be "required".

it's optional because it's either `app` (current) or `extension` (legacy)

Reply:
Gotcha. I think we should attempt to clean this all up a bit, but I suggest doing it as a follow up refactor, so as to unblock this PR.

Reply:
Yeah, sgtm